feat: recent stream activity.

zhyd1997 opened the pull request:
Fixes #393

zhyd1997 committed:
feat: recent stream activity.

mikeghen reviewed (changes requested):
I also think there's some missing icons from the frontend
<img width="1490" alt="Screen Shot 2022-03-03 at 10 06 25 AM" src="https://user-images.githubusercontent.com/5077880/156592130-097f0329-5801-4260-8ffc-cce79dcbf4ee.png">

mikeghen commented:
I left two comments, otherwise I think this is really good so thank you for taking a look at it @zhyd1997

bigint commented on package.json:
```diff
@@ -49,7 +50,8 @@
     "connected-react-router": "6.9.2",
     "cross-env": "7.0.3",
-    "ethers": "^5.5.4",
+    "ethers": "^5.5.2",
```
+1 if we downgrade this ENS name will not work 😕

zhyd1997 replied:
@mikeghen
not manually downgraded, maybe need update with v2 branch.

zhyd1997 commented:
> I also think there's some missing icons from the frontend <img alt="Screen Shot 2022-03-03 at 10 06 25 AM" width="1490" src="https://user-images.githubusercontent.com/5077880/156592130-097f0329-5801-4260-8ffc-cce79dcbf4ee.png">

I have added all token address listed in `polygon-config.js`,  could you tell me which token you're using that missed in this PR?

zhyd1997 committed:
chore: sync ethers library version.

zhyd1997 committed:
fix: util getTokenName.

zhyd1997 committed:
Merge branch 'v2' into issue-393

zhyd1997 commented:
Updated: f37fb3b

Try again and see if the token displays.

mikeghen reviewed (approved):
Very good! great addition

mikeghen commented:
@zhyd1997 there's one merge conflict, not sure what it is so have a look

zhyd1997 committed:
Merge branch 'v2' into issue-393

zhyd1997 commented:
Conflicts resolved!

zhyd1997 commented:
I need to regenerate fonts, give me 10 minutes.

zhyd1997 committed:
Merge branch 'v2' into issue-393

zhyd1997 committed:
fix: removed duplicated keys in i18n config file.

Seroxdesign commented:
This is really good! Great work

zhyd1997 commented:
@mikeghen If you have any questions about the PR, please let me know

zhyd1997 commented:
Additional Checklist:

- [x] default token image
- [x] query unlisted token name -> use `hyperlink` instead.
- [x] filter out zeroAddress (0x0000...)
- [x] support copy address

🚫  show streaming market if possible

@mikeghen @elvijsTDL

zhyd1997 committed:
fix: filtered out zero address for sent/recievied events.

zhyd1997 committed:
fix: set default token icon.
refactor: shorten assets path.

zhyd1997 committed:
feat: supported copy address.

zhyd1997 committed:
fix: added missing cancel stream.

zhyd1997 committed:
fix: corrected subscription and distribution publisher.

zhyd1997 committed:
fix: added missing MATICx address.

zhyd1997 committed:
fix: used react-tooltip for address.

zhyd1997 committed:
feat: added stop/start distribution copyings.

zhyd1997 committed:
fix: enhanced activity page mobile layout.
refactor: getTokenName can return undefined value for cleaning code.
feat: added unsupported token placeholder component.
fix: corrected subscription address.

zhyd1997 committed:
Merge branch 'Ricochet-Exchange:v2' into issue-393

0xSero commented:
This is really good!  Really nice to see how you used the superfluid sdk.

zhyd1997 commented:
Converting to draft PR again to consider if there a way to catch the stream that started from the Ricochet market.

zhyd1997 commented:
We can track started stream (or stop fixed stream) on our Launchpad according to the`flowsRecieved` and find it by this code:
https://github.com/Ricochet-Exchange/ricochet-frontend/blob/9a66f688f6eaef96a048e85cb59548594395e41d/src/store/main/selectors.ts#L7

just like the PR comments:
<img width="677" alt="PR 1130 comments" src="https://user-images.githubusercontent.com/31362988/156953912-bccbda53-859f-4151-965b-868cbe971922.png">

But we will miss it if the stream is stopped, so I tried my best and it's hard to track the market by stream information.

mikeghen commented:
This looks good @zhyd1997, I will merge it :+1:
